Configuration checklist (#7589)
fishtown-analytics/docs.getdbt.com: website/docs/docs/configuration-checklist.md

---
title: dbt platform configuration checklist
id: configuration-checklist
description: "Your to-do list for setting up your dbt platform account"
sidebar_label: "dbt platform configuration checklist"
pagination_next: null
pagination_prev: null
---

# dbt platform configuration checklist <Lifecycle status="self_service,managed,managed_plus" />

So, you've created a new cloud-hosted dbt platform account, and you're ready to explore its lightning-fast and intuitive features. Welcome! Before you begin, let’s ensure your account is properly configured so that you can easily onboard new users and take advantage of all the integrations dbt has to offer. 

For most organizations, this will require some collaboration with IT and/or security teams. Depending on the features you're using, you may need some of the following admin personas to help you get set up:
- Data warehouse (Snowflake, BigQuery, Databricks, etc.)
- Access control (Okta, Entra ID, Google, SAML 2.0)
- Git (GitHub, GitLab, Azure DevOps, etc.)

> This checklist is designed to ensure you have everything in the right place, allowing you to deploy quickly and without any bottlenecks. 

Reviewer: Is "bottleneck" widely understood or is it an idiom? If not, we might change it to "sources of friction" or "constraints"
```suggestion
This checklist ensures you have everything in the right place, allowing you to deploy quickly and without any bottlenecks. 
```

    - [ ] You can also verify the connection by running basic SQL queries using [dbt Insights](/docs/explore/access-dbt-insights).
- [ ] Create a single model and ensure that you can [run it](/reference/dbt-commands) successfully. 
    - [ ] For an easy to use drag-and-drop interface, try creating it with [dbt Canvas](/docs/cloud/canvas).
- [ ] Create a service account with proper access for your [production jobs](/docs/deploy/jobs).

## Git configuration

Git is, for many dbt environments, the backbone of your project. Git repositories are where your dbt files will live and where your developers will collaborate and manage version control of your project. 

- [ ] Configure a [Git repository](/docs/cloud/git/git-configuration-in-dbt-cloud) for your account. dbt supports integrations with:
    - [GitHub](/docs/cloud/git/connect-github)
    - [GitLab](/docs/cloud/git/connect-gitlab)
    - [Azure DevOps](/docs/cloud/git/connect-azure-devops)
    - Other providers using [Git clone](/docs/cloud/git/import-a-project-by-git-url)
    - If you aren't ready to integrate with an existing Git solution, dbt can provide you with a [managed Git repository](/docs/cloud/git/managed-repository). 

> - [ ] Developers are able to [checkout](/docs/cloud/git/version-control-basics#git-overview) a new branch in your repo.

Reviewer: ```suggestion
- [ ] Ensure developers can [checkout](/docs/cloud/git/version-control-basics#git-overview) a new branch in your repo.
```

> - [ ] Developers in the IDE can [commit changes](/docs/cloud/dbt-cloud-ide/ide-user-interface#basic-layout). 

Reviewer: ```suggestion
- [ ] Ensure developers in the IDE can [commit changes](/docs/cloud/dbt-cloud-ide/ide-user-interface#basic-layout). 
```

## Environments and jobs

> [Environments](/docs/environments-in-dbt) separate your development data from your production data. There are two types of environments - `Deployment` and `Development`. Within that distinction, there are three types of `Deployment` environments available on the dbt platform:

Author: ```suggestion
[Environments](/docs/environments-in-dbt) separate your development data from your production data. dbt supports two environment types: Development and Deployment. There are three types of Development environments:
```

> [Environments](/docs/environments-in-dbt) separate your development data from your production data. There are two types of environments - `Deployment` and `Development`. Within that distinction, there are three types of `Deployment` environments available on the dbt platform:

Reviewer: ```suggestion
[Environments](/docs/environments-in-dbt) separate your development data from your production data. dbt supports two environment types: Development (one per developer) and Deployment. Deployment environments include:
```

Author: Technically it's one development environment that's shared between all developers. I want to stick with the language we use elsewhere for describing this. If they navigate to Environments, they'll only see one `Development`

- Production - One per project
- Staging - One per project

> - General - As many as you need per project

Reviewer: ```suggestion
- General - Multiple per project
```

> Additionally, you will have only one `Development` environment per project, but each developer will have their own unique access to the IDE, separate from the work of other developers. 

Reviewer: much clearer!

[Jobs](/docs/deploy/jobs) dictate which commands are run in your environments and can be triggered manually, on a schedule, by other jobs, by APIs, or when pull requests are committed or merged. 

> Once your data warehouse is connected and Git integration is complete, it's time to configure environments and jobs:

Reviewer: Maybe make this active:
```suggestion
Once you connect your data warehouse and complete the Git integration, you can configure environments and jobs:
```

- [ ] Create a [Production Deployment environment](/docs/deploy/deploy-environments).
    - [ ] (Optional) Create an additional Staging or General environment.
- [ ] [Create and schedule](/docs/deploy/deploy-jobs#create-and-schedule-jobs) a deployment job. 
    - [ ] Validate the job by manually running it first.
- [ ] If needed, configure different databases for your environments.

## User access

The dbt platform offers a variety of access control tools that you can leverage to grant or revoke user access, configure RBAC, and assign user licenses and permissions.

- [ ] Manually [invite users](/docs/cloud/manage-access/invite-users) to the dbt platform, and they can authenticate using [MFA (SMS or authenticator app)](/docs/cloud/manage-access/mfa).
- [ ] Configure [single sign-on or OAuth](/docs/cloud/manage-access/sso-overview) for advanced access control. <Lifecycle status="managed,managed_plus" /> accounts only. 
    - [ ] Create [SSO mappings](/docs/cloud/manage-access/about-user-access#sso-mappings-) for groups
- Configure [System for Cross-Domain Identity Management (SCIM)](/docs/cloud/manage-access/scim) if available for your IdP. 

> - [ ] Users should be able to connect to the data warehouse from their personal profile. 

Reviewer: Not sure what the action is for them? Maybe:
```suggestion
- [ ] Ensure invited users are able to connect to the data warehouse from their personal profile. 
```

- [ ] [Create groups](/docs/cloud/manage-access/about-user-access#create-new-groups-) with granular permission sets assigned.
- [ ] Create [RBAC rules](https://docs.getdbt.com/docs/cloud/manage-access/about-user-access#role-based-access-control-) to assign users to groups and permission sets upon sign in. <Lifecycle status="managed,managed_plus" /> accounts only. 
- [ ] Enforce SSO for all non-admin users, and MFA is enforced for all password-based logins. 

## Continue the journey 

> Once you've completed this checklist, you're ready to start your dbt platform journey, but that journey has only just begun. You can explore these additional resources to support you along the way:

Author: ```suggestion
Once you've completed this checklist, you're ready to start your dbt platform journey, but that journey has only just begun. Explore these additional resources to support you along the way:
```

> - [ ] The [guides](/guides) offer quickstarts for warehouse connections and project features.

Author: ```suggestion
- [ ] Review the [guides](/guides) for quickstarts to help you get started with projects and features.
```

> - [ ] The [guides](/guides) offer quickstarts for warehouse connections and project features.

Reviewer: If you feel strongly about keeping the checklist, then I suggest telling them what they need to in each check list item. Are they _reviewing_ best practices? Are they _taking all_ of the dbt learn courses or taking specific ones? This feels more like a bullet list right now, but can easily be shifted to actions that require check marks. It comes off a little overwhelming because it's not specific.

> Once you've completed this checklist, you're ready to start your dbt platform journey, but that journey has only just begun. The docs in these pages will help you get the most out of what dbt has to offer, but there are other resources to aid you along the way:

Reviewer: ```suggestion
Once you've completed this checklist, you're ready to start your dbt platform journey, but that journey has only just begun. You can explore these additional resources to support you along the way:
```

> - [ ] [dbt Learn](https://learn.getdbt.com/catalog) has hands on courses and interactive content.

Author: ```suggestion
- [ ] Take a [dbt Learn](https://learn.getdbt.com/catalog) hands-on course.
```

> - [ ] The [guides](/guides) offer quickstarts for warehouse connections and project features.

Reviewer: These might not need to be checklist?

Author: I think it gives a consistent experience and allows them to check off if they've explored our docs

> - [ ] The [guides](/guides) offer quickstarts for warehouse connections and project features.

Reviewer: ```suggestion
- [ ] [Guides](/guides): Offer quickstarts for warehouse connections and project features.
```

Author: If we're going to take this approach, we should bold them in line with how we approach that elsewhere but if we leave them as checkboxes we can approach them in the same way we did with the rest of this page

> - [ ] Our [best practices](/best-practices) offer practical advice on structuring and deploying dbt projects.

Author: ```suggestion
- [ ] Review our [best practices](/best-practices) for practical advice on structuring and deploying your dbt projects.
```

> - [ ] [dbt Learn](https://learn.getdbt.com/catalog) has hands on courses and interactive content.

Reviewer: ```suggestion
- [ ] [dbt Learn](https://learn.getdbt.com/catalog): Hands on courses and interactive content.
```

> - [ ] The [references](/reference/references-overview) are the product dictionary and offer detailed implementation examples. 

Author: ```suggestion
- [ ] Become familiar with the [references](/reference/references-overview), as they are the product dictionary and offer detailed implementation examples. 
```

> - [ ] Our [best practices](/best-practices) offer practical advice on structuring and deploying dbt projects.

Reviewer: ```suggestion
- [ ] [Best practices](/best-practices): offer practical advice on structuring and deploying dbt projects.
```

> - [ ] The [references](/reference/references-overview) are the product dictionary and offer detailed implementation examples. 

Reviewer: ```suggestion
- [ ] [References](/reference/references-overview): Provide product dictionary and offer detailed implementation examples. 
```
